docker/install-dynamic-plugins.py

```diff
@@ -0,0 +1,116 @@
+import os
+import sys
+import yaml
+import tarfile
+import shutil
+import subprocess
+
+dynamicPluginsRoot = sys.argv[1]
+maxEntrySize = os.environ.get('MAX_ENTRY_SIZE', 10000000)
+
+dynamicPluginsFile = os.path.join(dynamicPluginsRoot, 'dynamic-plugins.yaml')
+dynamicPluginsGlobalConfigFile = os.path.join(dynamicPluginsRoot, 'app-config.dynamic-plugins.yaml')
+
+# test if file dynamic-plugins.yaml exists
+if not os.path.isfile(dynamicPluginsFile):
+    print(f'No {dynamicPluginsFile} file found. Skipping dynamic plugins installation.')
+    with open(dynamicPluginsGlobalConfigFile, 'w') as file:
+        file.write('')
+        file.close()
+    exit(0)
+
+with open(dynamicPluginsFile, 'r') as file:
+  plugins = yaml.safe_load(file)
+
+if plugins == '' or plugins is None:
+    print(f'{dynamicPluginsFile} file is empty. Skipping dynamic plugins installation.')
+    with open(dynamicPluginsGlobalConfigFile, 'w') as file:
+        file.write('')
+        file.close()
+    exit(0)
+
+class InstallException(Exception):
+    """Exception class from which every exception in this library will derive."""
+    pass
+
+# test that plugins is a list
+if not isinstance(plugins, list):
+    raise InstallException(f'{dynamicPluginsFile} content must be a list')
+
+globalConfig = {
+  'dynamicPlugins': {
+        'rootDirectory': 'dynamic-plugins-root'
+  }
+}
+
+def merge(source, destination):
+    for key, value in source.items():
+        if isinstance(value, dict):
+            # get node or create one
+            node = destination.setdefault(key, {})
+            merge(value, node)
+        else:
+            # if key exists in destination trigger an error
+            if key in destination:
+                raise InstallException('Config key ' + key + ' defined for 2 dynamic plugins')
+
+            destination[key] = value
+
+    return destination
+
+# iterate through the list of plugins
+for plugin in plugins:
+    package = plugin['package']
+    if package.startswith('./'):
+        package = os.path.join(os.getcwd(), package[2:])
+
+    print('\n======================= Installing dynamic plugin ' + package + '\n', flush=True)
+
+    # install the plugin
+    completed = subprocess.run(['npm', 'pack', package], capture_output=True, cwd=dynamicPluginsRoot)
+    if completed.returncode != 0:
+        raise InstallException(f'Error while installing plugin { package } with \'npm pack\' : ' + completed.stderr.decode('utf-8'))
+
+    archive = os.path.join(dynamicPluginsRoot, completed.stdout.decode('utf-8').strip())
+    directory = archive.replace('.tgz', '')
+    shutil.rmtree(directory, ignore_errors=True, onerror=None)
+    os.mkdir(directory)
+
+    # extract the plugin
+    file = tarfile.open(archive, 'r:gz')
```

Comment:
```suggestion
    file = tarfile.open(archive, 'r:gz') # NOSONAR
```

This should remove the security hotspot false positive

Reply:
Done. Thanks !